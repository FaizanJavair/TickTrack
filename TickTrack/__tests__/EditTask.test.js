import React from "react";
import renderer from "react-test-renderer";
import EditTaskModal from "../modals/editTaskModal";

list = {
  id: 1,
  name: "Drinking Water",
  color: "#414a4c",
  priority: "High",
  priorityValue: "2",
};
task = {
  id: "1",
  completed: "false",
  dueDateTime: { nano: "0", seconds: "1694214094" },
  title: "Drinking Water",
  color: "#414a4c",
  text: "06/09/2023 | 1:53 pm",
};

it("Edit Task Modal renders correctly", () => {
  const tree = renderer
    .create(<EditTaskModal task={task} list={list} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
