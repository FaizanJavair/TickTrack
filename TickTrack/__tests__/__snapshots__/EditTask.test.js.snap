// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Task Modal renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "left": "7%",
        "opacity": 1,
        "position": "absolute",
        "top": "7%",
      }
    }
  >
    <Text />
  </View>
  <View
    style={
      {
        "alignSelf": "stretch",
        "marginHorizontal": 40,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 30,
          "fontWeight": "600",
          "marginBottom": "5%",
          "textAlign": "center",
        }
      }
    >
      Edit Your Task
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholderTextColor="gray"
      style={
        {
          "borderColor": "#dadae8",
          "borderRadius": 12,
          "borderStyle": "solid",
          "borderWidth": 1.5,
          "height": 50,
          "paddingHorizontal": "2%",
        }
      }
      value="Drinking Water"
    />
    <View>
      <View
        style={
          {
            "flexDirection": "row",
            "marginTop": "3%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#414a4c",
              "borderRadius": 10,
              "fontSize": 15,
              "opacity": 1,
              "padding": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 16,
              }
            }
          >
            Due Date & Time
          </Text>
        </View>
        <Text
          style={
            {
              "fontSize": 18,
              "marginLeft": "2%",
              "marginTop": "2%",
            }
          }
        >
          06/09/2023 | 1:53 pm
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#414a4c",
          "borderRadius": 12,
          "height": 50,
          "justifyContent": "center",
          "marginTop": "6%",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 15,
            "fontWeight": "600",
          }
        }
      >
        Update
      </Text>
    </View>
  </View>
</View>
`;
